src/app/shared/services/auth.service.ts

```diff
@@ -1,37 +1,85 @@
 import { Injectable } from '@angular/core';
+import { Http, Response, Headers, RequestOptions } from '@angular/http';
 import { Observable, BehaviorSubject, ReplaySubject } from 'rxjs';
 
-import { ICredential } from '../../business-entities';
+import { ICredential, IToken, IUser } from '../../business-entities';
+import { AuthorizedHttp } from './authorized-http.service';
 
 const DELAY = 500;
 
 @Injectable()
 export class AuthService {
 
-    userInfo$: Observable<string>;
+    userInfo$: Observable<IUser>;
     isAuthenticated$: Observable<boolean>;
+    authToken: string;
 
+    private authServiceUrl: string = 'http://localhost:3004/auth';
     private isAuthenticated: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
-    private user: ReplaySubject<string> = new ReplaySubject<string>();
+    private user: ReplaySubject<IUser> = new ReplaySubject<IUser>();
 
-    constructor() {
+    constructor(private http: Http, private authorizedHttp: AuthorizedHttp) {
         this.userInfo$ = this.user.asObservable();
         this.isAuthenticated$ = this.isAuthenticated.asObservable();
     }
 
     login(userCredential: ICredential): Observable<boolean> {
         console.log(`login user:${userCredential.user} with password: ${userCredential.pass}`);
-        const clonedUser = Object.assign({}, userCredential);
+        const loginUrl = `${this.authServiceUrl}/login`;
+
+        return this.http
+            .post(loginUrl, {login: userCredential.user, password: userCredential.pass})
+            .map(this.mapData)
+            .map((res: IToken) => {
+                console.log('response', res);
+                if (res.token) {
+                    this.authToken = res.token;
```

Comment:
powinno sie unikac efektow ubocznych z wewnatrz strumienia (przypisywanie do this[prop])